src/app/views/marketplace/marketplace-app.controller.coffee

```diff
@@ -138,6 +146,61 @@ angular.module 'mnoEnterpriseAngular'
         )
 
       #====================================
+      # Comment modal
+      #====================================
+      vm.openCommentModal = (testimonial) ->
+        $uibModal.open(
+          templateUrl: 'app/views/marketplace/modals/comment-modal.html'
+          controller: 'CommentModal'
+          resolve:
+            testimonial: testimonial
+        )
+
+      #====================================
+      # Question modal
+      #====================================
+      vm.openQuestionModal = (questions) ->
+        $uibModal.open(
+          templateUrl: 'app/views/marketplace/modals/question-modal.html'
+          controller: 'QuestionModal'
+          resolve:
+            questions: questions
+        )
+
+      #====================================
+      # Answer modal
+      #====================================
+      vm.openAnswerModal = (question) ->
+        $uibModal.open(
+          templateUrl: 'app/views/marketplace/modals/answer-modal.html'
+          controller: 'AnswerModal'
+          resolve:
+            question: question
+        )
+
+      #====================================
+      # Edit modal
+      #====================================
+      vm.openEditModal = (object) ->
+        $uibModal.open(
+          templateUrl: 'app/views/marketplace/modals/edit-modal.html'
+          controller: 'EditModal'
+          resolve:
+            object: object
+        )
+
+      #====================================
+      # Deletion modal
+      #====================================
+      vm.openDeleteModal = (interactionObject) ->
```

Comment:
Instead of passing an object {key, object}, please use method signature
```
vm.openDeleteModal = (object, index) ->
```
Please take this comment in account everywhere this is done.